torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2115,6 +2118,7 @@ void ProcessGroupNCCL::watchdogHandler() {
       if (work.isCompleted()) {
         // Work status logging for desync debug
         desyncDebugger_.logWorkEnd(work);
+        FlightRecorder::get()->markEnd(work.trace_id_, work.getDuration());
```

Comment:
This is bad. Because when watchdog is hang, FR will dump inaccurate information because we didn't mark collectives to be "complete".

Reply:
Yeah, I understand this. I gave a trade-off analysis in "Re 2".

Reply:
I also added this: 
if NCCL comm is aborting, no "complete" information is reliable. Thus we prefer watchdog to explicitly mark entry as complete than FR doing so in the background or near the end of the program.

Reply:
Exactly, I consider it a higher bar to ask all accelerators to never hang in event query.